pkg/types/kustomization.go

```diff
@@ -136,3 +142,16 @@ type DataSources struct {
 	// i.e. a Docker .env file or a .ini file.
 	EnvSource string `json:"env,omitempty" yaml:"env,omitempty"`
 }
+
+// ImageTag contains an image and a new tag, which will replace the original tag
+type ImageTag struct {
+	// Name is the name for an image without tag
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+
+	// Tag will replace the original tag associated with Name at runtime
+	Tag string `json:"tag,omitempty" yaml:"tag,omitempty"`
+
+	// Env, if specified, the value in the environment variable will replace the original
+	// tag associated with Name at runtime
+	Env string `json:"env,omitempty" yaml:"env,omitempty"`
```

Comment:
i don't think we should support this.  It breaks accountability, in that one's configuration isn't entirely stored in a file.  If we allow env var override here, it suggests we should do so for nameprefix, commonlabel etc.

suggest instead we provide an "edit" command to easily change the NewTag field in a kustomization file.